### PR TITLE
artifactory-ha: Support for SSL termination in Nginx service layer. 

### PR DESCRIPTION
**Pull request by Satheesh-Balachandran**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
We currently don't have the support to terminate/offload SSL in the LoadBalancer layer of Nginx.

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/artifactory-ha/values.yaml [hunk 1]
```diff
@@ -1064,6 +1066,18 @@ nginx:
       if ($http_x_forwarded_proto = '') {
         set $http_x_forwarded_proto  $scheme;
       }
+      {{- if .Values.nginx.service.ssloffload }}
+      if ($host = $server_addr) {
+        rewrite ^/artifactory/webapp/(.*) http://$host:8081/artifactory/webapp/$1 redirect;
```
Comment (danielezer) [inline stable/artifactory-ha/values.yaml]: Change 8081 to `{{ .Values.artifactory.externalPort }}`

File: stable/artifactory-ha/values.yaml [hunk 2]
```diff
@@ -1064,6 +1066,18 @@ nginx:
       if ($http_x_forwarded_proto = '') {
         set $http_x_forwarded_proto  $scheme;
       }
+      {{- if .Values.nginx.service.ssloffload }}
+      if ($host = $server_addr) {
```
Comment (danielezer) [inline stable/artifactory-ha/values.yaml]: In what case would `$host` be different from the `$server_addr` ? From looking at the nginx docs, seems like they are the same. But maybe I'm missing something..?

File: stable/artifactory-ha/values.yaml [hunk 3]
```diff
@@ -1064,6 +1066,18 @@ nginx:
...
+        rewrite ^/artifactory/webapp/(.*) http://$host:8081/artifactory/webapp/$1 redirect;
+      }
+      if ($http_x_forwarded_proto != "https") {
+        rewrite ^/artifactory/webapp/(.*) https://$host/artifactory/webapp/$1 redirect;
+        rewrite ^/$ $scheme://$host/artifactory/webapp/#/home redirect;
+      }
+      if ($http_x_forwarded_proto = "https") {
+        rewrite ^/$ https://$host/artifactory/webapp/#/home redirect;
```
Comment (danielezer) [inline stable/artifactory-ha/values.yaml]: Doesn't this mean that every request that comes in with https will be redirected to the home page?

File: stable/artifactory-ha/values.yaml [hunk 4]
```diff
@@ -1085,13 +1099,25 @@ nginx:
...
+        {{- else }}
         proxy_set_header    X-Artifactory-Override-Base-Url $http_x_forwarded_proto://$host:$server_port/artifactory;
         proxy_set_header    X-Forwarded-Port  $server_port;
+        {{- end }}
         proxy_set_header    X-Forwarded-Proto $http_x_forwarded_proto;
         proxy_set_header    Host              $http_host;
         proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
       }
     }
+    {{- if .Values.nginx.service.ssloffload }}
```
Comment (danielezer) [inline stable/artifactory-ha/values.yaml]: Why is this needed?

Comment (Satheesh-Balachandran) [reply stable/artifactory-ha/values.yaml]: @eldada @danielezer If SSL is terminated in AWS load balancer with ACM certificate, only the HTTP request goes to the service.

File: stable/artifactory-ha/values.yaml (hunk 1 above)
Comment (Satheesh-Balachandran) [reply stable/artifactory-ha/values.yaml]: $server_addr is the address of the localhost in which Nginx runs. This is to make sure the Nginx readinessProbe/livenessProbe doesn't break. Nginx checks for artifactory health in https protocol. Another server block to listen in 8081 was configured which redirects to the application.

Comment (Satheesh-Balachandran) [reply stable/artifactory-ha/values.yaml]: @danielezer Suggest a better way to handle the Nginx readiness/liveness

File: stable/artifactory-ha/values.yaml (hunk 2 above)
Comment (Satheesh-Balachandran) [reply stable/artifactory-ha/values.yaml]: @danielezer Removed the Force SSL redirect part.

**Comment by eldada** [comment]
@Satheesh-Balachandran - can you wait for the new unified charts to apply this change there?

**Comment by Satheesh-Balachandran** [comment]
@eldada  Sure thanks!

**Comment by chukka** [comment]
@Satheesh-Balachandran  I'm closing this ,Feel free to raise a new PR for these changes
